### PR TITLE
FINERACT-2148: Zero interest charge off, interestRecalculation = false and fix installment`s update after zero interest charge-off transaction

### PR DESCRIPTION
**Pull request by mariiaKraievska**
## Description

Fix and add tests when interestRecalculation = false and zero interest charge-off behaviour. Also fix installment's update after zero interest charge-off transaction (skip zeroing interest for paid installments when interestRecalculation = false and fix last installment`s update).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java [hunk 1]
```diff
@@ -1213,40 +1212,50 @@ private void handleZeroInterestChargeOff(final LoanTransaction loanTransaction,
...
 
-            installments.stream().filter(installment -> installment.getFromDate().isAfter(transactionDate)).forEach(installment -> {
-                installment.updatePrincipal(MathUtil.nullToZero(installment.getPrincipal()).add(installment.getInterestCharged()));
-                installment.updateInterestCharged(BigDecimal.ZERO);
-            });
+            installments.stream()
+                    .filter(installment -> installment.getFromDate().isAfter(transactionDate) && !installment.isObligationsMet())
+                    .forEach(installment -> {
+                        final BigDecimal interestPaid = MathUtil.nullToZero(installment.getInterestPaid());
+                        final BigDecimal interestOutstanding = MathUtil.nullToZero(installment.getInterestCharged()).subtract(interestPaid);
```
Comment (adamsaghy) [inline fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java]: You might wanna consider using the `getInterestOutstanding` method which calculates properly the outstanding amount. TLDR: alongside the paid, it can be waived or written off as well...

Comment (mariiaKraievska) [reply fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java]: Done

**Review by adamsaghy — APPROVED** [review]
LGTM
